### DISCUSSION TITLE
Не подключается панель в Андроид приложении

### DISCUSSION THREAD

**Discussion by ltshmidt** [vvip-68/GyverPanelWiFi#286] [Задавайте вопросы]
Здравствуйте!
Невозможно подключиться к устройству через андроид приложение. Плата esp8266, после прошивки контроллер стартует, на кнопку отзывается, точку доступа WiFi поднимает. Телефон подключается к точке доступа успешно, но через приложение подключиться к устройству не получается. В тоже время, если подключиться к точке доступа с компьютера, то по API доступ и управление есть. В мониторе порта обмен идёт.
Подскажите, в чём может быть причина проблем с андроид-приложением?

**Comment by vvip-68** [comment]
Это не с андроид приложением проблема, а с телефоном, похоже...
Смотрите что запрещает пакетам уходить из телефона.
Поставьте какую-нибудь утилиту, которая пингует с телефона контроллер.
Смотрите настройки телефона проходит ли пинг. Как только добьетесь, что пинг с телефона на контроллер проходит - приложение тоже должно соединиться.

А чем вы через API с компьютера управляете?

**Reply by ltshmidt** [reply]
Здравствуйте!

С приложением разобрался. Проблема действительно в телефоне. Если отключить на нём передачу данных через мобильную связь, то приложение коннектится. Но возникла другая проблема. Когда в настройках приложения ввожу данные своей сети, то контроллер их почему-то не запоминает и после перезагрузки опять поднимает точку доступа.

Что касается управления по API, то тут всё просто. У меня на компе Ubuntu, я в терминале ввожу nc -u 192.168.4.1 2390 и дальше всё по вашему мануалу. В мониторе порта Arduino IDE контролирую выполнение команд.

**Reply by vvip-68** [reply]
Настройки подключения сохраняются в файловой системе микроконтроллера. Если при сборке проекта в опциях "Инструменты" выбран вариант распределения памяти FS:none - файловая система не будет создана - настройки некуда сохранять. Проверьте что у вас. Пришлите кусок лога, что выводится в мониторе порта при старте микроконтроллера.

**Reply by ltshmidt** [reply]
`20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: 'PanelAP
20:59:10.368 -> Точка доступа создана. Сеть: '1
20:59:10.368 -> 
20:59:10.368 -> UDP-сервер на порту 2390
20:59:10.376 -> Переход в авторежим отключен
20:59:10.376 -> Режим: Морской прибой`
Вот такое выводится при старте

**Reply by vvip-68** [reply]
Это какой-то странный вывод. Такого быть не должно.
Выводиться должно примерно следующее:
21:28:32.591 -> WiFiPanel v.1.14a.2023.1201
21:28:32.591 -> Контроллер: 'Wemos d1 mini'
21:28:32.591 -> Версия ядра: ESP8266 v3.1.2
21:28:32.591 -> FastLED: 3.6.0
21:28:32.591 -> Версия EEPROM: 0xA5
21:28:32.591 -> Host: 'WiFiPanel-0'
21:28:32.591 -> 
21:28:32.591 -> Инициализация файловой системы... 
21:28:32.917 -> Использовано 1990656 из 2072576 байт
21:28:32.917 -> 
21:28:32.964 -> 
21:28:32.964 -> Матрица: плоская 32x16
21:28:32.964 -> Адресация: по подключению
21:28:32.964 ->   Угол: левый верхний
21:28:32.964 ->   Направление: вниз
21:28:32.964 ->   Тип: зигзаг
21:28:32.964 -> Вывод на ленту:
21:28:32.964 ->   Линия 1 PIN=D2, START=0, COUNT=512
21:28:32.964 -> 
21:28:32.964 -> Доступные возможности:
21:28:32.964 -> + Язык интерфейса: RUS
21:28:32.964 -> + Бегущая строка: шрифт 5x8
21:28:32.964 -> + Кнопка управления: тактовая PIN=D1
21:28:32.964 -> + Синхронизация времени с сервером NTP
21:28:33.011 -> + Управление питанием PIN=D0
21:28:33.011 -> + Получение информации о погоде
21:28:33.011 -> + Управление через Web-канал
21:28:33.011 -> + Групповая синхронизация по протоколу E1.31
21:28:33.011 -> - Дополнительный индикатор TM1637
21:28:33.011 -> + Эффекты Jinx! с SD-карты CS=D8, CLK=D5, MISO=D6, MOSI=D7
21:28:33.011 -> + Поддержка MP3 Player STX=D3 SRX=D4
21:28:33.011 -> 
21:28:33.011 -> Выделение памяти для LEDS   : 28296 -> 26752 -- 1544
21:28:33.011 -> Выделение памяти для OVERLAY: 26752 -> 25208 -- 1544

**Reply by ltshmidt** [reply]
Сейчас вот так:
`l⸮bn⸮n⸮⸮b⸮⸮nn'l⸮l`⸮nnl`nr⸮⸮⸮nbl⸮`p⸮n⸮r⸮bbn⸮nb⸮⸮nn'l`⸮nnl`nr⸮⸮⸮nb⸮`r⸮⸮nb⸮`⸮ll`⸮⸮n⸮`⸮⸮n⸮r⸮⸮n|⸮l⸮l`bbrl⸮nb⸮nl`⸮rl⸮l⸮l`⸮rl⸮l⸮
WiFiPanel v.1.13.2023.1111
Версия EEPROM: 0xFF
Обновлено до: 0x5D
Host: 'WiFiPanel'

Инициализация файловой системы... 
Использовано 16384 из 2072576 байт

Инициализация EEPROM...
Сохранение строк выполнено.
Использованы адреса EEPROM 1000 - 1037
Свободно ячеек 3058
Настройки сохранены в EEPROM


Матрица: плоская 12x12
Адресация: по подключению
  Угол: левый верхний
  Направление: вниз
  Тип: зигзаг

Доступные возможности:
+ Бегущая строка: шрифт 5x8
+ Кнопка управления: тактовая
+ Синхронизация времени с сервером NTP
- Управление питанием
+ Получение информации о погоде
- Управление по каналу MQTT
- Групповая синхронизация по протоколу E1.31
- Дополнительный индикатор TM1637
- Эффекты Jinx! с SD-карты
- Поддержка MP3 Player

Поиск файлов карт индексов...
Нет файлов карт индексов

sMATRIX_TYPE=0

Найденные анимации:
   Сердце	8x8

Поиск слайдов на FS в папке '12p12'...
Поиск слайдов на FS в папке '16p16'...
Слайды не найдены. Будут использованы картинки эффекта 'Погода'

Создание точки доступа PanelAP
Точка доступа создана. Сеть: 'PanelAP'. Пароль: '12341234'.
IP адрес: 192.168.4.1

UDP-сервер на порту 2390
Режим: Дождь
Настройки сохранены в EEPROM
`

**Reply by vvip-68** [reply]
Вот сейчас - правильный вывод.
Из которого видно, что файловая система создается, значит место для хранения настроек определено.
Но это не дает ответ на вопрос почему не сохраняется введенный пароль и настройки сетевого подключения.
Я не знаю. Вы их точно настраиваете в приложении и сохраняете?

**Reply by ltshmidt** [reply]
Всё. С сетью разобрался. Спасибо за поддержку!
Остался один вопрос. После включения постоянно светит светодиод на плате контроллера. Как его погасить?

**Reply by vvip-68** [reply]
Это значит, что вы назначили вывод сигнала на ленту на вывод D4 мкроконтроллера.
При рекомендованных настройках скетча (параметры по умолчанию, установлено правильное (рекомендуемое) ядро микроконтроллера) вывод на ленту назначается на вывод D2 контроллера. Тогда диод на плате не горит.

Разбирайтесь что сделали не так или зачем переназначили вывод с D2 на D4.

**Reply by ltshmidt** [reply]
Спасибо огромное!

**Reply by gornostayi** [reply]
Здравствуйте! 
Столкнулся с проблемой - в андройд приложении не появляются иконки на верхней панели для настройки  эффектов матрицы и прочего. Имеются только четыре стандартные - соединение, настройки матрицы, настройки синхронизации, настройки подключения.

**Reply by vvip-68** [reply]
Там три андроид приложения. Каждое для своего функционала. Вы описали приложение для НАСТРОЙКИ матрицы, подключения и т.д. Для УПРАВЛЕНИЯ матрицей (эффекты,тексты, часу и тюд) - установите другое приложение.

**Reply by gornostayi** [reply]
Благодарю!!! PanelSetupWiFi - для настройки сети. GyverPanelWiFi - для управления матрицой (лентой).
